inspirehep/modules/workflows/views.py

```diff
@@ -591,16 +613,71 @@ def put(self):
         return jsonify(data), 200
 
 
+class ResolveEditArticleResource(MethodView):
+    """Resolve `edit_article` callback.
+
+    When the workflow needs to resolve conficts, the workflow stops in
+    ``HALTED`` state, to continue this endpoint is called. If it's called
+    and the conflicts are not resolved it will just save the workflow.
+
+    Args:
+        workflow_data (dict): the workflow object send in the request's payload.
+
+    """
+
+    def put(self):
+        """Handle callback for merge conflicts."""
+        workflow_data = workflow_loader()
+        workflow_id = workflow_data['id']
+
+        try:
+            workflow = workflow_object_class.get(workflow_id)
+        except WorkflowsMissingObject:
+            raise CallbackWorkflowNotFoundError(workflow_id)
+
+        if workflow.status != ObjectStatus.WAITING or \
+           'callback_url' not in workflow.extra_data:
+            raise CallbackWfNotInWaitingForCurationState(workflow.id)
+
+        workflow_id = workflow.id
+        workflow.data = workflow_data['metadata']
+        workflow.status = ObjectStatus.RUNNING
+        workflow.extra_data.pop('callback_url', None)
+        workflow.save()
+        db.session.commit()
+        workflow.continue_workflow(delayed=True)
+        data = {'message': 'Workflow {} is continuing.'.format(workflow_id)}
+        return jsonify(data), 200
+
+
+def start_edit_article_workflow(rec_id):
+    record = get_db_record('lit', rec_id)
+    eng_uuid = start('edit_article', data=record)
+    obj_id = WorkflowEngine.from_uuid(eng_uuid).objects[0].id
+    url = "{}{}".format(current_app.config['WORKFLOWS_EDITOR_API_URL'], obj_id)
+    return redirect(location=url, code=302)
+
+
 callback_resolve_validation = ResolveValidationResource.as_view(
     'callback_resolve_validation')
 callback_resolve_merge_conflicts = ResolveMergeResource.as_view(
     'callback_resolve_merge_conflicts')
+callback_resolve_edit_article = ResolveEditArticleResource.as_view(
+    'callback_resolve_edit_article')
 
-blueprint.add_url_rule(
+callback_blueprint.add_url_rule(
     '/workflows/resolve_validation_errors',
     view_func=callback_resolve_validation,
 )
-blueprint.add_url_rule(
+callback_blueprint.add_url_rule(
     '/workflows/resolve_merge_conflicts',
     view_func=callback_resolve_merge_conflicts,
 )
+workflow_blueprint.add_url_rule(
+    '/jlab_edit/<rec_id>',
```

Comment:
I wouldn't make this jlab specific, as it might be useful more generally, even if it's only used by them at first.

Reply:
Now that we clarified that the permissions for this are the same as the ones to edit any record, we can easily do that yep, also for the callback

Reply:
:+1: agree

Reply:
btw, any suggestions?

Reply:
Why not `/edit/literature/<rec_id>`? in the future, when the editor can handle other collections, we can then generalize this to `/edit/<collection>/<rec_id>`.

Reply:
hmm...., something like `/edit_article/<rec_id>`? (seeing that the workflow is called edit_article)